### PR TITLE
feat(docs): Reproducible Test Case Page

### LINKED ISSUE #11933: [docs] add page in contributing section on building reproducible builds
## Summary

A notice when filing a bug report here is to create a demo project to show off the bug. I think it would be useful to either enhance the "How to file an issue" page or create a new page describing how to create a reproducible build and the benefits of it.

### Motivation

It would be good to have a page describing further how to make a demo project for such.

### PR DESCRIPTION
**Pull request by lannonbr**
## Description

Wrote a page on having some content on describing reproducible Test Cases, how to make one, and why they are useful

## Related Issues

Resolves #11933.

### CODE REVIEW

File: docs/contributing/how-to-make-a-reproducible-build.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+- Start up a Gatsby site with a starter.
+- Add the plugins specific to the issue. (If you are having something go wrong with gatsby-MDX, load in the related MDX plugins, but don't load in the other plugins you don't need that may also be in your actual site)
+- Add the code to show the error
+- Push up the code and then link it when [creating an issue](/contributing/how-to-file-an-issue/).
+
+## Places to develop a reproducible build
+
+- Locally with a starter: We can start out with a [Starter](/docs/starters) locally and then build it on your own machine.
+- Host on CodeSandbox: CodeSandbox has a [Gatsby template](https://codesandbox.io/s/github/gatsbyjs/gatsby-starter-default) which you can develop a site up online without needing to download anything. Then you can link a URL to the project and have a live application of your issue all in the browser.
```
Comment (bryndyment) [inline docs/contributing/how-to-make-a-reproducible-build.md]: `which you can develop a site up online` -> `allowing you to host a Gatsby project online`

File: docs/contributing/how-to-make-a-reproducible-build.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+One way to try to dig down at resolving an issue is to have a demo project that shows off just the singular issue at hand. If you are able to, you could link the repo that the bug is found in, but it is easier to tackle the bug with a reproducible build instead so you have a lot less surface area to tackle.
+
+A reproducible build is a reproduction of a problem in a much smaller context so other developers can reproduce the bug in a more narrow codebase rather than an entire application.
+
+## Steps to create a reproducible build
+
+- Start up a Gatsby site with a starter.
+- Add the plugins specific to the issue. (If you are having something go wrong with gatsby-MDX, load in the related MDX plugins, but don't load in the other plugins you don't need that may also be in your actual site)
```
Comment (bryndyment) [inline docs/contributing/how-to-make-a-reproducible-build.md]: `(If you are having …` -> `(E.g., if you are having …`

**Comment by m-allanson** [comment]
That or "Creating a Bug Reproduction"?

(Some other ideas: Create a bug demo project, Create an issue demo, Creating a bug reproduction, How to demo a bug, Building an issue example, Demonstrate a problem... nothing jumps out as definitely the right answer)

**Comment by m-allanson** [comment]
@lannonbr suggested "Reduced Test Case" in Discord, that seems like a good fit to me. The doc title could then be "Creating a Reduced Test Case".

**Comment by lannonbr** [comment]
Cool. I can rephrase the content of the page to align with that terminology.

**Comment by DSchau** [comment]
Bike shedding here (🚴‍♂️🚴‍♂️🚴‍♂️) but I’d probably say “Creating a Reproducible Test Case” and merge the ideas here! But happy with the proposed suggestions too!

**Comment by KyleAMathews** [comment]
Very helpful doc!

**Comment by krismorf** [comment]
How about using `npx gatsby ...` throughout the doc to ensure latest cli version is used?

**Comment by lannonbr** [comment]
Updated the terminology to "Reproducible Test Case" and took @krismorf's suggestion and added npx in the doc.

**Comment by LekoArts** [comment]
We don’t want to use npx 🙆🏼‍♀️
https://github.com/gatsbyjs/gatsby/issues/12195

**Comment by m-allanson** [comment]
We've gone back and forth on `npx`, we're going to standardise on not using it - see issue 
#12195 linked by @LekoArts for more details on that decision. 

Thanks @lannonbr, merging 🎉

**Pull request merged**